session-8/model/defining/task1.go

```diff
@@ -0,0 +1,30 @@
+package defining
+
+import (
+	"math"
+)
+
+type Shape interface {
+	Area() float64
+}
+
+type Circle struct {
+	Radius float64
+}
+
+type Rectangle struct {
+	Width  float64
+	Height float64
+}
+
+func (c Circle) Area() float64 {
+	return math.Pi * c.Radius * c.Radius
+}
+
+func (r Rectangle) Area() float64 {
+	return r.Height * r.Width
+}
+
+func AreaOfShape(shape Shape) interface{} {
+	return shape.Area()
+}
```

Comment:
interface{} əvəzinə `any` də işlətmək olar. Amma bu bizə gələcəkdə yazacağımız testlərdə mane ola bilər. Area() float64 return verdiyinə görə, AreaOfShape() də həmişə float64 qaytaracaq. Ona görə mən burda float64 return etməyin tərəfdarıyam
```suggestion
func AreaOfShape(shape Shape) float64 {
	return shape.Area()
}
```